DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -1490,30 +1490,32 @@ public async Task<DiscordMember> GetMemberAsync(ulong userId, bool updateCache =
         CancellationToken cancellationToken = default
     )
     {
-        int recievedLastCall = 1000;
         ulong last = 0ul;
-        while (recievedLastCall > 0)
+        for (int i = MemberCount; i > 0;)
         {
             if (cancellationToken.IsCancellationRequested)
             {
                 yield break;
             }
-            
+
             IReadOnlyList<TransportMember> tms = await this.Discord.ApiClient.ListGuildMembersAsync(this.Id, 1000, last == 0 ? null : last);
```

Comment:
While out of scope for this PR, please rename this to `members` or `transportMembers`